client/my-sites/plan-features-2023-grid/components/billing-timeframe.tsx

```diff
@@ -88,12 +98,91 @@ function usePerMonthDescription( {
 	return null;
 }
 
+interface WooExpressMonthlyPromotionProps {
+	billingTimeframe: TranslateResult;
+	monthlyPlanSlug: string;
+	yearlyPlanSlug: string;
+}
+
+const DiscountPromotion = styled.div`
+	text-transform: uppercase;
+	font-weight: 600;
+	color: #306e27;
+	font-size: $font-body-extra-small;
+	margin-top: 6px;
+`;
+
+const WooExpressMonthlyPromotion: FunctionComponent< WooExpressMonthlyPromotionProps > = (
+	props
+) => {
+	const translate = useTranslate();
+	const { billingTimeframe, yearlyPlanSlug, monthlyPlanSlug } = props;
+
+	const annualPlan = getPlans()[ yearlyPlanSlug ];
+	const monthlyPlan = getPlans()[ monthlyPlanSlug ];
+	const planPrices = useSelector( ( state ) => ( {
+		annualPlanMonthlyPrice: getPlanRawPrice( state, annualPlan.getProductId(), true ) || 0,
+		monthlyPlanPrice: getPlanRawPrice( state, monthlyPlan.getProductId() ) || 0,
+	} ) );
```

Comment:
We could have used the [getPlanPrices](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/plans/selectors/get-plan-prices.ts) selector for these instead. `rawPrice` from there would correspond to `getPlanRawPrice`. It's another means to unify the pricing selectors, at least until we have a better data layer.

Reply:
Hey @chriskmnds, we'll use the `usePerMonthDescription` to get the description once #76535 is merged. My intention is to have something similar to: 
```ts
const WooExpressMonthlyPromotion: FunctionComponent< WooExpressMonthlyPromotionProps > = (
	props
) => {
	const { billingTimeframe, discountDescription } = props;
	return (
		<div>
			<div>{ billingTimeframe }</div>
			<DiscountPromotion>{ discountDescription }</DiscountPromotion>
		</div>
	);
};
```

Reply:
Cool. I guess let's rebase this PR over https://github.com/Automattic/wp-calypso/pull/76535?

There's also a bit of refactoring happening in https://github.com/Automattic/wp-calypso/pull/74797 with this file, which reinvigorates a data store here. It's been approved, but mainly waiting for the WooExpress slugs to settle in, as they are needed in its [base branch](https://github.com/Automattic/wp-calypso/pull/76086).

Reply:
Hey! I rebased as you suggested and updated the code accordingly. I checked the changes on the file, and I believe it won't affect my changes in this PR.